packages/driver/src/httpScram.ts

```diff
@@ -41,27 +66,42 @@ export async function HTTPSCRAMAuth(
       Authorization: `SCRAM-SHA-256 data=${utf8ToB64(clientFirst)}`,
     },
   });
-  if (serverFirstRes.status === 403) {
-    // tslint:disable-next-line:no-console
-    console.log(serverFirstRes);
-    throw new Error(`Server doesn't support HTTP SCRAM authentication`);
+
+  // The first request must have status 401 Unauthorized and provide a
+  // WWW-Authenticate header with a SCRAM-SHA-256 challenge.
+  // See: https://github.com/edgedb/edgedb/blob/09782afd3b759440abbb1b26ee19b6589be04275/edb/server/protocol/auth/scram.py#L153-L157
+  const authenticateHeader = serverFirstRes.headers.get("WWW-Authenticate");
+  if (serverFirstRes.status !== 401 || !authenticateHeader) {
+    const body = await serverFirstRes.text();
+    throw new ProtocolError(`authentication failed: ${body}`);
   }
-  const firstAttrs = parseHeaders(serverFirstRes.headers, "WWW-Authenticate");
-  if (firstAttrs.size === 0) {
-    throw new Error("Invalid credentials");
+
+  // WWW-Authenticate can contain multiple comma-separated authentication
+  // schemes (each with own comma-separated parameter pairs), but we only support
+  // one SCRAM-SHA-256 challenge, e.g., `SCRAM-SHA-256 sid=..., data=...`.
+  if (!authenticateHeader.startsWith("SCRAM-SHA-256")) {
+    throw new ProtocolError(
+      `unsupported authentication scheme: ${authenticateHeader}`
+    );
   }
-  if (!firstAttrs.has("sid") || !firstAttrs.has("data")) {
+
+  // The server may respond with a 401 Unauthorized and `WWW-Authenticate: SCRAM-SHA-256` with
+  // no parameters if authentication fails, e.g., due to an incorrect username.
+  // See: https://github.com/edgedb/edgedb/blob/09782afd3b759440abbb1b26ee19b6589be04275/edb/server/protocol/auth/scram.py#L112-L120
+  const authParams = authenticateHeader.split(/ (.+)?/, 2)[1] ?? "";
+  if (authParams.length === 0) {
+    const body = await serverFirstRes.text();
+    throw new ProtocolError(`authentication failed: ${body}`);
+  }
+
+  const { sid, data: serverFirst } = parseSidAndData(authParams);
+  if (!sid || !serverFirst) {
     throw new ProtocolError(
-      `server response doesn't contain '${
-        !firstAttrs.has("sid") ? "sid" : "data"
-      }' attribute`
+      `authentication challenge missing '${!sid ? "sid" : "data"}' attribute`
```

Comment:
Falsy is a reasonable check in the `if` conditional, but I think it might be helpful to log the stringified value of both attributes here since `null` vs `""` might be helpful when debugging.